app/src/main/java/com/randomappsinc/studentpicker/grouping/GroupMakingSettingsViewHolder.java

```diff
@@ -70,4 +75,35 @@ private void applyNumberOfGroups() {
         }
         numGroups.clearFocus();
     }
+
+    @Override
+    public void beforeTextChanged(CharSequence s, int start, int count, int after) {}
+
+    @Override
+    public void onTextChanged(CharSequence s, int start, int before, int count) {}
+
+    @Override
+    public void afterTextChanged(Editable editable) {
+        if (isEditableAndFocused(editable, namesPerGroup)) {
+            numGroups.setText(getOffset(namesPerGroup));
+        } else if (isEditableAndFocused(editable, numGroups)) {
+            namesPerGroup.setText(getOffset(numGroups));
+        }
+    }
+
+    private String getOffset(EditText editText) {
+        String inputtedNumber = editText.getText().toString().trim();
+        int newNumber = (inputtedNumber.isEmpty()) ? 0 :
+                Integer.parseInt(inputtedNumber);
+        if (newNumber <= 0) {
+            newNumber = editText.getContext().getResources()
+                    .getInteger(R.integer.default_number_of_names_per_group);
+        }
+        int offset = (int) Math.ceil((double) settings.getNameListSize() / newNumber);
+        return String.valueOf(offset);
+    }
+
+    private boolean isEditableAndFocused(Editable editable, EditText editText) {
+        return editable == editText.getEditableText() && editText.isFocused();
```

Comment:
Ah, this is the struggle of binding to 2 different EditTexts with the same listener. With ButterKnife, we will be able to have 2 separate and custom named methods to listen to these callbacks. However, we will need to prevent infinite chaining. So if we change the text in number of groups, we will automatically change the text in names per group. When the callback fires for names per group, we will just early return since this EditText is not in focus.